ios/AppLovinMAX.m

```diff
@@ -2266,12 +2267,33 @@ - (void)setAmazonResult:(id /* DTBAdResponse or DTBAdErrorInfo */)result forAdUn
     if ( adFormat == MAAdFormat.interstitial )
     {
         MAInterstitialAd *interstitial = [self retrieveInterstitialForAdUnitIdentifier: adUnitIdentifier];
-        [interstitial setLocalExtraParameterForKey: key value: result];
+        
+        if (interstitial)
+        {
+            [interstitial setLocalExtraParameterForKey: key value: result];
+        }
+        else
+        {
+            [self log: @"Failed to set Amazon result - unable to find interstitial"];
+        }
     }
-    else
+    else  // MAAdFormat.banner or MAAdFormat.mrec
     {
-        MAAdView *adView = [self retrieveAdViewForAdUnitIdentifier: adUnitIdentifier adFormat: adFormat];
-        [adView setLocalExtraParameterForKey: key value: result];
+        MAAdView *adView = [AppLovinMAXAdView sharedWithAdUnitIdentifier: adUnitIdentifier];
+
+        if ( !adView )
+        {
+            adView = [self retrieveAdViewForAdUnitIdentifier: adUnitIdentifier adFormat: adFormat];
+        }
+        
+        if ( adView )
+        {
+            [adView setLocalExtraParameterForKey: key value: result];
+        }
+        else
+        {
+            [self log: @"Failed to set Amazon result - unable to find %@", [[adFormat label] lowercaseString]];
```

Comment:
You can just print `adFormat`. It automatically calls `description` which is fine, that's what we're doing on Android which automatically calls `toString()` (you're not calling `getLabel()`).